### PR TITLE
Primitive types improvements 

### PR DESCRIPTION
**Pull request by draganjovanovic1**
TryParse where possible e.g. DateTime and DateTimeOffset, uint32 added, small improvements

### CODE REVIEW

**Comment by MangelMaxime** [comment]
Hello @draganjovanovic1 thank you for the PR :)

I will need time to digest it and think about it because some of the limitations was on purpose. Especially, the ones about the dates because we had different behavior depending on the machine language.

**Comment by draganjovanovic1** [comment]
Hi @MangelMaxime,

I understand that my PR is quite big. I actually wanted to change only one thing and ended up changing whole bunch of them :) Sorry for messy PR. In the end I realized that it would be better idea to open an issue and discuss it first but since I already wrote some code I sent it. Hope you don't mind.

Let me try to explain my motivation behind it. Actually dates were my main concern. I didn't like the idea of concatenating ISO date manually when it is supported in the framework type (see [this](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#Roundtrip)). Actually `date.ToString("O")` was used on the `System.DateTime` type already but not on the `System.DateTimeOffset`.  If your concern is decoding and you want to force ISO-8601 as only valid format maybe we could use `DateTimeOffset.TryParseExact` instead of regex.

I am open for discussion. Cheers.

**Comment by MangelMaxime** [comment]
@draganjovanovic1 Don't worry we don't have that many contribution on Thoth.Json so it's fine to start the conversation with a PR. :)

The problematic part is really the `DateTime` and `DateTimeOffet` changes which are making the tests fails on non english computer. If I understand correctly how .Net works, dependings on the computer language (or something like that) its supporting a different `parse. format` for the Date.

File: src/Thoth.Json/Decode.fs [hunk 1]
```diff
@@ -251,32 +264,31 @@ module Decode =
             if Helpers.isNumber value then
                 Helpers.asFloat value |> decimal |> Ok
             elif Helpers.isString value then
-                Helpers.asString value |> System.Decimal.Parse |> Ok
+                try
+                    Helpers.asString value |> System.Decimal.Parse |> Ok
```
[the remark this answers is not in the document]
Comment (draganjovanovic1) [reply src/Thoth.Json/Decode.fs]: Already tried, but unfortunately it is not possible: `FABLE: Cannot resolve System.Decimal.TryParse`

**Comment by MangelMaxime** [comment]
Using `CultureInfo.InvariantCulture` make the tests pass :) 

Thank you for the PR.

**Comment by MangelMaxime** [comment]
This has been released in version `2.2.0`

**Pull request merged**